.github/workflows/buildandtest.yml

```diff
@@ -22,6 +22,8 @@ jobs:
         run: dotnet restore
       - name: Build
         run: dotnet build --no-restore
+      - name: Format
+        run: dotnet format --verify-no-changes --severity=error
```

Comment:
Optional / non-blocking : should we do some sort of version pinning for the `dotnet format` version and mention that in the README too?

Reply:
dotnet format is a part of dotnet (starting from 6). that is why we don't even install it in the CI. so there is no need to lock it, because we already lock the dotnet version